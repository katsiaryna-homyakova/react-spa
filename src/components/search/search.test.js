import React from 'react';
import { render, shallow, mount } from 'enzyme';
import Search from './search';

describe('Search component', () => {
  it('renders correctly', () => {
    const component = render(<Search />);
    expect(component).toMatchSnapshot();
  });

  it('handle input change', () => {
    const component = shallow(<Search />);
    component.find('input').simulate('change', { target: { value: 'foo' } });
    expect(component.state('value')).toEqual('foo');
  });

  it('handle filter change', () => {
    const component = mount(<Search />);
    component
      .find('.filter button')
      .at(1)
      .simulate('click');
    expect(component.state('activeFilter')).toEqual('genre');
  });
});
