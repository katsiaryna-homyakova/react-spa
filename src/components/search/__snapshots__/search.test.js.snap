// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "search",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "title",
        },
        "children": Array [
          Object {
            "data": "Find your movie",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "input-group",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "placeholder": "Search",
                "type": "text",
                "value": "",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "btn base-button pink",
                  "id": "submitSearch",
                },
                "children": Array [
                  Object {
                    "data": " Search",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "btn base-button pink",
                "id": "submitSearch",
              },
              "children": Array [
                Object {
                  "data": " Search",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "placeholder": "Search",
                  "type": "text",
                  "value": "",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "filter",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "filter-title",
                },
                "children": Array [
                  Object {
                    "data": "Search by",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "btn base-button active",
                  },
                  "children": Array [
                    Object {
                      "data": " Title",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "btn base-button ",
                    },
                    "children": Array [
                      Object {
                        "data": " Genre",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "button",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "btn base-button active",
                },
                "children": Array [
                  Object {
                    "data": " Title",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "btn base-button ",
                  },
                  "children": Array [
                    Object {
                      "data": " Genre",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "filter-title",
                  },
                  "children": Array [
                    Object {
                      "data": "Search by",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "btn base-button ",
                },
                "children": Array [
                  Object {
                    "data": " Genre",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "btn base-button active",
                  },
                  "children": Array [
                    Object {
                      "data": " Title",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "filter-title",
                    },
                    "children": Array [
                      Object {
                        "data": "Search by",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "span",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "input-group",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "placeholder": "Search",
              "type": "text",
              "value": "",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "btn base-button pink",
                "id": "submitSearch",
              },
              "children": Array [
                Object {
                  "data": " Search",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "btn base-button pink",
              "id": "submitSearch",
            },
            "children": Array [
              Object {
                "data": " Search",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "placeholder": "Search",
                "type": "text",
                "value": "",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "id": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "id": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "filter",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "filter-title",
              },
              "children": Array [
                Object {
                  "data": "Search by",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "btn base-button active",
                },
                "children": Array [
                  Object {
                    "data": " Title",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "btn base-button ",
                  },
                  "children": Array [
                    Object {
                      "data": " Genre",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "btn base-button active",
              },
              "children": Array [
                Object {
                  "data": " Title",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "btn base-button ",
                },
                "children": Array [
                  Object {
                    "data": " Genre",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "filter-title",
                },
                "children": Array [
                  Object {
                    "data": "Search by",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "btn base-button ",
              },
              "children": Array [
                Object {
                  "data": " Genre",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "btn base-button active",
                },
                "children": Array [
                  Object {
                    "data": " Title",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "filter-title",
                  },
                  "children": Array [
                    Object {
                      "data": "Search by",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "title",
          },
          "children": Array [
            Object {
              "data": "Find your movie",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "filter",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "filter-title",
            },
            "children": Array [
              Object {
                "data": "Search by",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "btn base-button active",
              },
              "children": Array [
                Object {
                  "data": " Title",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "btn base-button ",
                },
                "children": Array [
                  Object {
                    "data": " Genre",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "btn base-button active",
            },
            "children": Array [
              Object {
                "data": " Title",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "btn base-button ",
              },
              "children": Array [
                Object {
                  "data": " Genre",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "filter-title",
              },
              "children": Array [
                Object {
                  "data": "Search by",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "btn base-button ",
            },
            "children": Array [
              Object {
                "data": " Genre",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "btn base-button active",
              },
              "children": Array [
                Object {
                  "data": " Title",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "filter-title",
                },
                "children": Array [
                  Object {
                    "data": "Search by",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "input-group",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "placeholder": "Search",
                "type": "text",
                "value": "",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "btn base-button pink",
                  "id": "submitSearch",
                },
                "children": Array [
                  Object {
                    "data": " Search",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "btn base-button pink",
                "id": "submitSearch",
              },
              "children": Array [
                Object {
                  "data": " Search",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "placeholder": "Search",
                  "type": "text",
                  "value": "",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "title",
            },
            "children": Array [
              Object {
                "data": "Find your movie",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
